tests/unit/test_table.py

```diff
@@ -5993,6 +5994,71 @@ def test_external_catalog_table_options_from_api_repr(self):
         assert result == expected
 
 
+class TestForeignTypeInfo:
```

Comment:
I'd also like to see a test where we do `Table.from_api_repr` and `Table.to_api_repr` so that we can visually compare that the correct `schema.foreignTypeInfo` field of the REST API object is set.

Reply:
Not yet addressed.

Reply:
Done.